In [173]:
import pandas as pd

### 1. concat 데이터 불러오기

297243 rows × 168 columns

In [174]:
df = pd.read_csv('../외감데이터/dataset_외감.csv',index_col=0)

In [175]:
df = df[df['회계년도'].str.endswith('12')]

In [176]:
df

회사명   거래소코드     회계년도 감사의견 감사의견코드   산업코드                 산업명  \
0          (유)가든주류   97082  2019/12  NaN    NaN  74603      음ㆍ식료품 및 담배 도매업   
1          (유)가든주류   97082  2020/12  NaN     QS  74603      음ㆍ식료품 및 담배 도매업   
2          (유)가든주류   97082  2021/12  NaN     UQ  74603      음ㆍ식료품 및 담배 도매업   
3          (유)가든주류   97082  2022/12  NaN     DU  74603      음ㆍ식료품 및 담배 도매업   
4       (유)그룹세브코리아   13171  2019/12  NaN    NaN  74604            생활용품 도매업   
...            ...     ...      ...  ...    ...    ...                 ...   
307858  주식회사씨앤케이무역  101008  2021/12  NaN     UQ  74603      음ㆍ식료품 및 담배 도매업   
307859  주식회사씨앤케이무역  101008  2022/12  NaN     UQ  74603      음ㆍ식료품 및 담배 도매업   
307860  주식회사씨앤케이스틸   69534  2013/12  NaN    NaN  32402         1차 비철금속 제조업   
307861  주식회사씨앤케이스틸   69534  2014/12  NaN     UQ  32402         1차 비철금속 제조업   
307862  주식회사씨앤케이스틸   69534  2015/12  NaN     UQ  32402         1차 비철금속 제조업   

        업종코드  소속코드         설립일  ... [제조]인건비(백만원) [제조]금융비용(백만원)  [제조]임차료(백만원)  \
0          1     7  1995/04/21  ...      3665.94        627.90         41.21   
1          1     7  1995/04/21  ...      3850.72        476.63          7.64   
2          1     7  1995/04/21  ...      3221.61        467.05         20.43   
3          1     7  1995/04/21  ...      3059.12       1913.99         35.93   
4          1     7  1997/08/01  ...      8006.40        378.86       3557.11   
...      ...   ...         ...  ...          ...           ...           ...   
307858     1     7  2016/02/04  ...       992.43        218.93         50.00   
307859     1     7  2016/02/04  ...      1019.78        313.37         40.63   
307860     1     7  2012/08/30  ...        13.69        -51.65          4.97   
307861     1     7  2012/08/30  ...       282.55       1212.43         20.93   
307862     1     7  2012/08/30  ...       411.31       1209.75         22.05   

       [제조]조세공과(백만원)  [제조]감가상각비(백만원) [제조]종업원1인당 부가가치(백만원).1  [제조]총자본투자효율.1  \
0             133.24          592.23                    0.0          21.20   
1              50.56          358.23                    0.0          16.22   
2              81.76          217.94                    0.0          16.63   
3             123.39          555.08                    0.0          11.59   
4             306.51          785.52                    0.0          48.63   
...              ...             ...                    ...            ...   
307858        114.43          784.99                    0.0          24.31   
307859         88.14          160.47                    0.0          10.53   
307860          1.04            0.00                    0.0           1.77   
307861         13.64          261.04                    0.0          -4.92   
307862         39.05         1434.26                    0.0          -4.26   

        [제조]기계투자효율.1  [제조]부가가치율.1  [제조]종업원수  
0             445.21        13.95       0.0  
1             343.62        12.12       0.0  
2            2995.65        12.89       0.0  
3             750.05         3.48       0.0  
4               0.00        13.37       0.0  
...              ...          ...       ...  
307858        248.93        24.50       0.0  
307859        116.23        12.40       0.0  
307860          0.00        88.84       0.0  
307861         -9.13       -68.41       0.0  
307862         -6.60       -16.23       0.0  

[297243 rows x 168 columns]

### 2. 금융업종 제외하기

산업명 : 금융, 부동산 임대 및 공급업

회사명 : 투자회사, 부동산투자회사, 부동산, 금융투자주식회사, 선박투자회사, 투자주식회사, 투자개발, 금융투자

투자 포함된 회사 검색 후 삭제 (주)건설투자, (주)국제선박투자운용 등등...
리츠 포함된 회사 검색 후 삭제

257351 rows × 168 columns

In [177]:
# 산업 업종 분류가 잘못된 데이터 고려
# 투자, 리츠 포함된 회사는 따로 찾아서 검색 후 삭제

df= df[~df['산업명'].str.contains('금융')]
df= df[~df['산업명'].str.contains('부동산 임대 및 공급업')]
df= df[~df['회사명'].str.contains('투자회사')]
df= df[~df['회사명'].str.contains('부동산투자회사')]
df= df[~df['회사명'].str.contains('부동산')]
df= df[~df['회사명'].str.contains('금융투자주식회사')]
df= df[~df['회사명'].str.contains('선박투자회사')]
df= df[~df['회사명'].str.contains('투자회사')]
df= df[~df['회사명'].str.contains('투자주식회사')]
df= df[~df['회사명'].str.contains('투자개발')]
df= df[~df['회사명'].str.contains('금융투자')]

In [178]:
# 투자 삭제할 회사명 리스트

companies_to_remove = [
    '(주)건설투자', '(주)국제선박투자운용', '주식회사아프로투자',
    '주식회사인천투자펀드2', '주식회사케이티투자운용', '주식회사테라투자파트너스', 
    '(주)케이리츠투자운용','케이디기술투자(주)', '케이엔씨개발투자컨설팅', '(유)비티투자',
    '신동해투자일호피에프브이(주)', '아이비투자검단(주)', '에이스투자(주)', 
    '유한회사한일자산관리앤투자','주식회사건인투자', '주식회사세원투자산업'
]

# 데이터프레임에서 해당 회사명을 가진 행들을 제거

df = df[~df['회사명'].isin(companies_to_remove)]

In [179]:
# 리츠 삭제할 회사명 리스트

companies_to_remove = ['굿모닝대신리츠주식회사', '주식회사평안리츠',  '(주)청리츠앤파트너스',
    '주식회사힘내라리츠', '케이리츠주식회사', '대영리츠건설(주)', '(주)리츠산업',
    '(주)리츠홀딩스',  '신한리츠운용주식회사',
    '월드리츠개발주식회사', '주식회사경성리츠', '주식회사글리츠리츠', '주식회사리츠파크', '주식회사무등리츠',
    '주식회사비앤에스리츠']

# 데이터프레임에서 해당 회사명을 가진 행들을 제거

df = df[~df['회사명'].isin(companies_to_remove)]

### 3. 결측치가 너무 많은 데이터 삭제 (5000개 이상), 컬럼명 통일

257351 rows × 135 columns

In [180]:
# 필요 없는 컬럼제거

df.drop(columns=['상장일','상장폐지일','업종코드','소속코드','감사의견','[제조]종업원1인당 부가가치(백만원)','[제조]종업원1인당 매출액(백만원)',
                '[제조]종업원1인당 경상이익 (2007년 이전 발생)(백만원)','[제조]종업원1인당 순이익(백만원)','[제조]종업원1인당 인건비(백만원)',
                '[제조]노동장비율','[제조]기계장비율','[제조]자본집약도','[제조]원','[제조]재공품회전률','[제조]종업원수','[제조]종업원1인당 부가가치(백만원).1',
                '[제조]총자본투자효율.1','[제조]기계투자효율.1','[제조]부가가치율.1','[제조]부가가치(백만원).1','[제조]경상이익증가율 (2007년 이전 발생)',
                '[제조]매출액경상이익률 (2007년 이전 발생)','[제조]총자본경상이익률 (2007년 이전 발생)','[제조]자기자본경상이익률 (2007년 이전 발생)',
                '[제조]자본금경상이익률 (2007년 이전 발생)','[제조]기업경상이익률 (2007년 이전 발생)','[제조]투자수익률 (2007년 이전 발생)','[제조]종업원1인당 부가가치증가율', 
                '[제조]종업원수증가율', '[제조]종업원1인당 매출액증가율', '[제조]종업원1인당 인건비증가율','[제조]1주당경상이익 (2007년 이전 발생)(원)'],inplace=True)

In [181]:
# 컬럼명 가독성

df.columns=[col.replace('(백만원)','') for col in df.columns]
df.columns=[col.replace('(원)','') for col in df.columns]
df.columns=[col.replace('[제조]','') for col in df.columns]

### 3. 재무비율 생성을 위해 2년 연속 재무제표가 존재하지 않으면 제거

255099 rows × 135 columns

In [182]:
# 회계년도를 시간 데이터로 변경
df['회계년도'] = pd.to_datetime(df['회계년도'], format='%Y/%m')

# 거래소코드와 회계년도 기준으로 정렬
df = df.sort_values(by=['거래소코드', '회계년도'])

# 3년 연속 데이터 체크
def has_consecutive_years(group):
    # year 만 가져오기
    years = group['회계년도'].dt.year
    # 현재 year에서 전 yeear을 뺀값이 1, 총합이 2이상인 것을 반환
    return (years.diff().fillna(1) == 1).cumsum().max() >= 2

# 거래소코드를 기준으로 groupby 후 filter
df = df.groupby('거래소코드').filter(has_consecutive_years)

In [183]:
df

회사명   거래소코드       회계년도 감사의견코드    산업코드  \
93850           (주)포스코건설   10000 2012-12-01     UQ   64102   
93851           (주)포스코건설   10000 2013-12-01     UQ   64102   
93852           (주)포스코건설   10000 2014-12-01     UQ   64102   
93853           (주)포스코건설   10000 2015-12-01     UQ   64102   
93854           (주)포스코건설   10000 2016-12-01     UQ   64102   
...                  ...     ...        ...    ...     ...   
283300      주식회사더퍼스트에스제이  107424 2022-12-01     DU  199609   
295121        주식회사부성개발산업  107453 2021-12-01    NaN  126802   
295122        주식회사부성개발산업  107453 2022-12-01     DS  126802   
54961   주식회사코차이나로지스틱스코리아  107471 2021-12-01    NaN   85209   
54962   주식회사코차이나로지스틱스코리아  107471 2022-12-01     UQ   85209   

                       산업명         설립일  기업규모코드 기업규모명  \
93850               토목 건설업  1994/12/01    10.0   대기업   
93851               토목 건설업  1994/12/01    10.0   대기업   
93852               토목 건설업  1994/12/01    10.0   대기업   
93853               토목 건설업  1994/12/01    10.0   대기업   
93854               토목 건설업  1994/12/01    10.0   대기업   
...                    ...         ...     ...   ...   
283300      그 외 기타 개인 서비스업  2019/08/19     0.0   NaN   
295121         부동산 관련 서비스업  2019/12/16     NaN   NaN   
295122         부동산 관련 서비스업  2019/12/16     0.0   NaN   
54961         기타 운송관련 서비스업  2009/01/19     NaN   NaN   
54962         기타 운송관련 서비스업  2009/01/19     0.0   NaN   

        통계청 한국표준산업분류 코드 10차(대분류)  ...  부가가치율  노동소득분배율  자본분배율  이윤분배율  \
93850                         41  ...   8.40    16.93  83.07  58.52   
93851                         41  ...   3.64    39.42  60.58  33.82   
93852                         41  ...   2.94    54.85  45.15   6.04   
93853                         41  ...   3.48    50.03  49.97  13.35   
93854                         41  ... -12.97     0.00   0.00   0.00   
...                          ...  ...    ...      ...    ...    ...   
283300                        96  ...   0.00     0.00   0.00   0.00   
295121                        68  ...  25.80     0.00   0.00  86.11   
295122                        68  ...  20.31     0.00   0.00  74.51   
54961                         52  ...  11.40    42.23  57.77  37.72   
54962                         52  ...   2.75    73.30  26.70  -5.90   

        법인세비용차감전(계속사업)손익        인건비      금융비용       임차료     조세공과     감가상각비  
93850          458072.46  100145.17   -419.33  11489.42  4456.67  17680.02  
93851          155753.76  115047.15 -14196.96  11115.02  5042.92  19116.94  
93852           35583.76  121320.85  21074.13  13187.29  5165.64  24839.25  
93853           55211.14  113666.26  14255.74  10594.08  5933.40  27515.47  
93854         -858337.57  167421.94 -64787.70  12881.66  5619.91  24626.08  
...                  ...        ...       ...       ...      ...       ...  
283300         -11034.69     431.68   5225.38    904.78    72.50     52.52  
295121           2760.89       0.00    428.62      8.00     8.66      0.00  
295122           3719.02       0.00    -13.73      0.00    58.02      0.00  
54961            1358.69    1209.55     62.70     78.93    42.12    111.94  
54962             -92.53     730.80    179.37     42.96    26.08    110.37  

[255099 rows x 135 columns]

### 4. Labeling을 위해 자본잠식률 추가

자본잠식률, 이자보상배율 추가

253037 rows × 160 columns

In [184]:
plus_f = pd.read_csv('../외감데이터/rate_final.csv')

In [185]:
plus_f['회계년도'] = pd.to_datetime(plus_f['회계년도'], format='%Y/%m')

In [186]:
df = pd.merge(df, plus_f, on=['회사명','거래소코드','회계년도'], how='inner')


### 5. Labeling 진행

감사의견코드 : 'DI', 'DS', 'DU', 'AG', 'QS'

    부도: A
    정상: a

부도 : 17829

자본잠식률 : 100%

    부도: B
    정상: b

부도 : 24035

자본잠식률 : 50% (2년 연속)

    부도: C
    정상: c

부도 : 20051

In [187]:
# 회계년도를 datetime
df['회계년도'] = pd.to_datetime(df['회계년도'], format='%Y-%m-%d')
df = df.sort_values(by=['거래소코드', '회계년도'])

# 감사의견코드 중 DI, DS DU, AG, QS가 존재하면 A 아니면 a
df['감사의견_y'] = df['감사의견코드'].apply(lambda x: "A" if x in ['DI', 'DS', 'DU', 'AG', 'QS'] else "a")


In [188]:
# 회계년도를 datetime
df['회계년도'] = pd.to_datetime(df['회계년도'], format='%Y-%m-%d')
df = df.sort_values(by=['거래소코드', '회계년도'])

df['자본잠식률_100'] = df['자본잠식률'].apply(lambda x: "B" if x >= 100 else "b")

In [189]:
# 데이터셋을 거래소코드와 회계년도 기준으로 정렬
df.sort_values(by=['거래소코드', '회계년도'], inplace=True)

# 새로운 'Label' 컬럼을 기본값 'c'로 추가
df['자본잠식률_50'] = 'c'

# 이전 행의 값 초기화
prev_code = None
prev_cap_erode_rate = None

# 각 행을 순회하며 조건에 따라 'Label' 컬럼 설정
for index, row in df.iterrows():
    current_code = row['거래소코드']
    current_cap_erode_rate = row['자본잠식률']

    # 현재 행과 이전 행이 같은 회사이고, 두 해 모두 자본잠식률이 50 이상인 경우 'C'로 표시
    if current_code == prev_code and current_cap_erode_rate >= 50 and prev_cap_erode_rate >= 50:
        df.at[index, '자본잠식률_50'] = 'C'

    # 이전 행 값 업데이트
    prev_code = current_code
    prev_cap_erode_rate = current_cap_erode_rate

In [190]:
# 라벨링 조건에 따라 'y' 컬럼 생성
df['y'] = ((df['감사의견_y'] == 'A') | 
        (df['자본잠식률_100'] == 'B') | 
        (df['자본잠식률_50'] == 'C')).astype(int)

In [191]:
df[(df['y'] == 0)]

회사명   거래소코드       회계년도 감사의견코드    산업코드               산업명  \
0               (주)포스코건설   10000 2012-12-01     UQ   64102            토목 건설업   
1               (주)포스코건설   10000 2013-12-01     UQ   64102            토목 건설업   
2               (주)포스코건설   10000 2014-12-01     UQ   64102            토목 건설업   
3               (주)포스코건설   10000 2015-12-01     UQ   64102            토목 건설업   
4               (주)포스코건설   10000 2016-12-01     UQ   64102            토목 건설업   
...                  ...     ...        ...    ...     ...               ...   
253029            주식회사마커  107421 2021-12-01    NaN  126802       부동산 관련 서비스업   
253030            주식회사마커  107421 2022-12-01     UQ  126802       부동산 관련 서비스업   
253033        주식회사부성개발산업  107453 2021-12-01    NaN  126802       부동산 관련 서비스업   
253035  주식회사코차이나로지스틱스코리아  107471 2021-12-01    NaN   85209      기타 운송관련 서비스업   
253036  주식회사코차이나로지스틱스코리아  107471 2022-12-01     UQ   85209      기타 운송관련 서비스업   

               설립일  기업규모코드 기업규모명  통계청 한국표준산업분류 코드 10차(대분류)  ...  new_자기자본증가율  \
0       1994/12/01    10.0   대기업                        41  ...          NaN   
1       1994/12/01    10.0   대기업                        41  ...     2.842294   
2       1994/12/01    10.0   대기업                        41  ...     0.143478   
3       1994/12/01    10.0   대기업                        41  ...    14.401493   
4       1994/12/01    10.0   대기업                        41  ...   -23.096221   
...            ...     ...   ...                       ...  ...          ...   
253029  2013/01/11     NaN   NaN                        68  ...          NaN   
253030  2013/01/11     0.0   NaN                        68  ...  1083.062264   
253033  2019/12/16     NaN   NaN                        68  ...          NaN   
253035  2009/01/19     NaN   NaN                        52  ...          NaN   
253036  2009/01/19     0.0   NaN                        52  ...   -24.271852   

        new_유형자산증가율  new_유동자산증가율  new_재고자산증가율  new_매출액증가율  new_비유동자산증가율  \
0               NaN          NaN          NaN         NaN           NaN   
1         40.756393   -18.258059   -16.164419   14.016854      4.081387   
2          5.344684   -12.449193    -8.092776   -6.396890     17.867219   
3         -4.509898     6.351912   -34.770304  -13.012219     -1.042386   
4        -10.676382    -5.085581    54.677241  -15.921085     -8.134513   
...             ...          ...          ...         ...           ...   
253029          NaN          NaN          NaN         NaN           NaN   
253030   112.134784   -26.218401  -100.000000  -33.930305    105.289701   
253033          NaN          NaN          NaN         NaN           NaN   
253035          NaN          NaN          NaN         NaN           NaN   
253036    -3.789171    58.495828          NaN   44.162036     -3.705472   

        감사의견_y  자본잠식률_100  자본잠식률_50  y  
0            a          b         c  0  
1            a          b         c  0  
2            a          b         c  0  
3            a          b         c  0  
4            a          b         c  0  
...        ...        ...       ... ..  
253029       a          b         c  0  
253030       a          b         c  0  
253033       a          b         c  0  
253035       a          b         c  0  
253036       a          b         c  0  

[211543 rows x 164 columns]

### 6. 라벨링 후에 첫행 제거

211123 rows × 164 columns

In [160]:
# 거래소코드를 기준으로 그룹바이, 첫행 삭제
grouped_df = df.groupby('거래소코드').apply(lambda x: x.iloc[1:]).reset_index(drop=True)

### 7. 부도년도 이후 데이터 삭제

164099 rows × 164 columns

In [161]:
def remove_post_default_data(group):
    # 최초 부도 년도 설정
    default_year = group[group['y'] == 1]['회계년도'].min()
    
    # default 값이 nan이 아니면 작거나 같은 데이터만 가져옴
    if not pd.isnull(default_year):
        group = group[group['회계년도'] <= default_year]
    
    return group

# 회사명으로 그룹바이 후 함수 적용
df = grouped_df.groupby('거래소코드').apply(remove_post_default_data).reset_index(drop=True)

### 8. 부도년도 이전 데이터

전년도의 자본잠식률이 50%면 1
나머지 데이터는 삭제

153832 rows × 164 columns

In [162]:
# 거래소 코드와 회계년도를 기준으로 정렬
df = df.sort_values(by=['거래소코드', '회계년도'])

for i in range(1, len(df)):
    # y가 1 인 경우
    if df.loc[i, 'y'] == 1:
        # 해당 행과 전 행의 거래소 코드가 같으면
        if df.loc[i, '거래소코드'] == df.loc[i-1, '거래소코드']:
            # 전 행의 자본잠식률이 50이상이면
            if df.loc[i-1, '자본잠식률'] >= 50:
                # 전 행의 y 가 1
                df.loc[i-1, 'y'] = 1

In [163]:
# 부도 데이터 중 정상 데이터 삭제

def remove_post_default_data(group):
    # 최초 부도 년도 설정
    default_year = group[group['y'] == 1]['회계년도'].min()
    
    # default 값이 nan이 아니면 크거나 같은 데이터만 가져옴
    if not pd.isnull(default_year):
        group = group[group['회계년도'] >= default_year]
    
    return group

# 회사명으로 그룹바이 후 함수 적용
df = df.groupby('거래소코드').apply(remove_post_default_data).reset_index(drop=True)

### 9. 중간 데이터 확인

#### 전체 153832


#### 정상 134189


#### 부도 19643

    감사의견 : 12000

    자본잠식률100% : 7212

    자본잠식률50% : 5970



153832 rows × 164 columns

In [103]:
df.to_csv('../외감데이터/dataset_부도11_첫행날림.csv')

In [169]:
df[(df['자본잠식률_100']=='B')&(df['자본잠식률_50']=='C')]

회사명   거래소코드       회계년도 감사의견코드    산업코드                 산업명  \
128         대한상선주식회사   10023 2013-12-01     UQ   85001              해상 운송업   
378         프라임개발(주)   10071 2013-12-01     UQ   64101              건물 건설업   
389       동양인터내셔널(주)   10079 2013-12-01     UQ   74607           기타 전문 도매업   
403       인천공항에너지(주)   10086 2013-12-01     UQ   43501                 전기업   
561          (주)대동주택   10211 2013-12-01     DU   64102              토목 건설업   
...              ...     ...        ...    ...     ...                 ...   
153817      주식회사리컴퍼니  107340 2022-12-01     QS   74604            생활용품 도매업   
153822       (주)바로대흥  107384 2022-12-01     DU   64101              건물 건설업   
153823  카일라스프로퍼티주식회사  107386 2022-12-01     UQ  126800                부동산업   
153826     주식회사곰할인마트  107401 2022-12-01     DS   74701              종합 소매업   
153829  주식회사더퍼스트에스제이  107424 2022-12-01     DU  199609      그 외 기타 개인 서비스업   

               설립일  기업규모코드 기업규모명  통계청 한국표준산업분류 코드 10차(대분류)  ...  new_자기자본증가율  \
128     1983/09/01    30.0  중견기업                        50  ...   -11.052900   
378     1988/08/18    30.0  중견기업                        41  ...    46.608914   
389     1995/04/01    30.0  중견기업                        46  ...   189.214175   
403     1997/05/19    10.0   대기업                        35  ...    -1.842373   
561     1987/03/04    20.0  중소기업                        41  ...     6.613695   
...            ...     ...   ...                       ...  ...          ...   
153817  2012/03/23     0.0   NaN                        46  ...   -79.610080   
153822  2021/07/08     0.0   NaN                        41  ...    98.163135   
153823  2021/03/01     0.0   NaN                        68  ... -2662.706368   
153826  2018/01/24     0.0   NaN                        47  ...    26.613933   
153829  2019/08/19     0.0   NaN                        96  ...   116.358174   

        new_유형자산증가율  new_유동자산증가율  new_재고자산증가율  new_매출액증가율  new_비유동자산증가율  \
128      -26.359149     0.576240    -8.904584  -15.568921    -20.937577   
378       -6.580639   -23.318986   -42.466302  -80.684824     -7.123525   
389      -33.031214   -45.664452  -100.000000  -95.698223    -88.958698   
403        5.153690   -27.617644   -30.937499   -5.694189     -1.227897   
561      -84.093771    -2.560674    -0.557795  -98.519991    -41.173578   
...             ...          ...          ...         ...           ...   
153817    71.153458    37.231630    -1.430780  -15.867632     65.280185   
153822   -60.724179   -10.239169     1.388026         NaN     -2.643182   
153823          NaN    -7.850192     3.701481 -100.000000     -0.626867   
153826    -0.213036   -51.625853  -100.000000  -69.497540     -0.672948   
153829    -0.741997    -2.328674     2.531782         NaN     13.007582   

        감사의견_y  자본잠식률_100  자본잠식률_50  y  
128          a          B         C  1  
378          a          B         C  1  
389          a          B         C  1  
403          a          B         C  1  
561          A          B         C  1  
...        ...        ...       ... ..  
153817       A          B         C  1  
153822       A          B         C  1  
153823       a          B         C  1  
153826       A          B         C  1  
153829       A          B         C  1  

[4495 rows x 164 columns]

In [116]:
df[df['y']==0]

회사명   거래소코드       회계년도 감사의견코드    산업코드  \
0               (주)포스코건설   10000 2013-12-01     UQ   64102   
1               (주)포스코건설   10000 2014-12-01     UQ   64102   
2               (주)포스코건설   10000 2015-12-01     UQ   64102   
3               (주)포스코건설   10000 2016-12-01     UQ   64102   
4               (주)포스코건설   10000 2017-12-01     UQ   64102   
...                  ...     ...        ...    ...     ...   
153820         주식회사신현대관리  107373 2022-12-01     UQ  147401   
153821    주식회사훈민정음엔터테이먼트  107374 2022-12-01     UQ  137309   
153825       주식회사어라운드테이블  107400 2022-12-01     UQ  126802   
153828            주식회사마커  107421 2022-12-01     UQ  126802   
153831  주식회사코차이나로지스틱스코리아  107471 2022-12-01     UQ   85209   

                                산업명         설립일  기업규모코드 기업규모명  \
0                            토목 건설업  1994/12/01    10.0   대기업   
1                            토목 건설업  1994/12/01    10.0   대기업   
2                            토목 건설업  1994/12/01    10.0   대기업   
3                            토목 건설업  1994/12/01    10.0   대기업   
4                            토목 건설업  1994/12/01    10.0   대기업   
...                             ...         ...     ...   ...   
153820              사업시설 유지·관리 서비스업  2009/01/20     0.0   NaN   
153821      그 외 기타 전문, 과학 및 기술 서비스업  2013/07/25     0.0   NaN   
153825                  부동산 관련 서비스업  2021/10/21     0.0   NaN   
153828                  부동산 관련 서비스업  2013/01/11     0.0   NaN   
153831                 기타 운송관련 서비스업  2009/01/19     0.0   NaN   

        통계청 한국표준산업분류 코드 10차(대분류)  ...  new_자기자본증가율  new_유형자산증가율  new_유동자산증가율  \
0                             41  ...     2.842294    40.756393   -18.258059   
1                             41  ...     0.143478     5.344684   -12.449193   
2                             41  ...    14.401493    -4.509898     6.351912   
3                             41  ...   -23.096221   -10.676382    -5.085581   
4                             41  ...     1.397914     6.969776   -11.399067   
...                          ...  ...          ...          ...          ...   
153820                        74  ...    -6.677929   636.548907    19.360597   
153821                        73  ...   149.955579   -33.852589   413.622098   
153825                        68  ...   -40.763510          NaN  3247.443594   
153828                        68  ...  1083.062264   112.134784   -26.218401   
153831                        52  ...   -24.271852    -3.789171    58.495828   

        new_재고자산증가율  new_매출액증가율  new_비유동자산증가율  감사의견_y  자본잠식률_100  자본잠식률_50  y  
0        -16.164419   14.016854      4.081387       a          b         c  0  
1         -8.092776   -6.396890     17.867219       a          b         c  0  
2        -34.770304  -13.012219     -1.042386       a          b         c  0  
3         54.677241  -15.921085     -8.134513       a          b         c  0  
4        -19.424219   14.943191     36.635257       a          b         c  0  
...             ...         ...           ...     ...        ...       ... ..  
153820          NaN   30.580983    503.346734       a          b         c  0  
153821          NaN  238.928734    -32.918536       a          b         c  0  
153825          inf         inf    -88.591692       a          b         c  0  
153828  -100.000000  -33.930305    105.289701       a          b         c  0  
153831          NaN   44.162036     -3.705472       a          b         c  0  

[134189 rows x 164 columns]

In [117]:
df

회사명   거래소코드       회계년도 감사의견코드    산업코드  \
0               (주)포스코건설   10000 2013-12-01     UQ   64102   
1               (주)포스코건설   10000 2014-12-01     UQ   64102   
2               (주)포스코건설   10000 2015-12-01     UQ   64102   
3               (주)포스코건설   10000 2016-12-01     UQ   64102   
4               (주)포스코건설   10000 2017-12-01     UQ   64102   
...                  ...     ...        ...    ...     ...   
153827         어반이엔씨주식회사  107410 2022-12-01     DS   64101   
153828            주식회사마커  107421 2022-12-01     UQ  126802   
153829      주식회사더퍼스트에스제이  107424 2022-12-01     DU  199609   
153830        주식회사부성개발산업  107453 2022-12-01     DS  126802   
153831  주식회사코차이나로지스틱스코리아  107471 2022-12-01     UQ   85209   

                       산업명         설립일  기업규모코드 기업규모명  \
0                   토목 건설업  1994/12/01    10.0   대기업   
1                   토목 건설업  1994/12/01    10.0   대기업   
2                   토목 건설업  1994/12/01    10.0   대기업   
3                   토목 건설업  1994/12/01    10.0   대기업   
4                   토목 건설업  1994/12/01    10.0   대기업   
...                    ...         ...     ...   ...   
153827              건물 건설업  2013/07/17     0.0   NaN   
153828         부동산 관련 서비스업  2013/01/11     0.0   NaN   
153829      그 외 기타 개인 서비스업  2019/08/19     0.0   NaN   
153830         부동산 관련 서비스업  2019/12/16     0.0   NaN   
153831        기타 운송관련 서비스업  2009/01/19     0.0   NaN   

        통계청 한국표준산업분류 코드 10차(대분류)  ...  new_자기자본증가율  new_유형자산증가율  new_유동자산증가율  \
0                             41  ...     2.842294    40.756393   -18.258059   
1                             41  ...     0.143478     5.344684   -12.449193   
2                             41  ...    14.401493    -4.509898     6.351912   
3                             41  ...   -23.096221   -10.676382    -5.085581   
4                             41  ...     1.397914     6.969776   -11.399067   
...                          ...  ...          ...          ...          ...   
153827                        41  ...  -100.109324   146.455199     4.898104   
153828                        68  ...  1083.062264   112.134784   -26.218401   
153829                        96  ...   116.358174    -0.741997    -2.328674   
153830                        68  ...   130.494239          NaN     0.868438   
153831                        52  ...   -24.271852    -3.789171    58.495828   

        new_재고자산증가율  new_매출액증가율  new_비유동자산증가율  감사의견_y  자본잠식률_100  자본잠식률_50  y  
0        -16.164419   14.016854      4.081387       a          b         c  0  
1         -8.092776   -6.396890     17.867219       a          b         c  0  
2        -34.770304  -13.012219     -1.042386       a          b         c  0  
3         54.677241  -15.921085     -8.134513       a          b         c  0  
4        -19.424219   14.943191     36.635257       a          b         c  0  
...             ...         ...           ...     ...        ...       ... ..  
153827    53.629708  -80.067273     38.713607       A          B         c  1  
153828  -100.000000  -33.930305    105.289701       a          b         c  0  
153829     2.531782         NaN     13.007582       A          B         C  1  
153830   -95.809449   49.131823      0.000000       A          b         c  1  
153831          NaN   44.162036     -3.705472       a          b         c  0  

[153832 rows x 164 columns]